contracts/eth/nearbridge/contracts/NearDecoder.sol

```diff
@@ -30,12 +33,20 @@ library NearDecoder {
     struct BlockProducer {
         PublicKey publicKey;
         uint128 stake;
+        // Flag indicating if this validator proposed to be a chunk-only producer (i.e. cannot become a block producer).
+        bool isChunkOnly;
     }
 
     function decodeBlockProducer(Borsh.Data memory data) internal pure returns (BlockProducer memory res) {
+        uint8 validator_version = data.decodeU8();
         data.skipBytes();
         res.publicKey = data.decodePublicKey();
         res.stake = data.decodeU128();
+        if (validator_version == VALIDATOR_V2) {
+            res.isChunkOnly = data.decodeU8() != 0;
```

Comment:
`res.isChunkOnly = data.decodeU8() != 0;` This line will always produce `true`.

Reply:
why?

Reply:
If the data is not 0, return true. If it is 0, return false. How can that always produce true?

Reply:
@joshuajbouw yeah, I'm also confused how this could be always `true`. Maybe I'm missing something.
@0x3bfc please comment

Reply:
Sorry I was confused with the value of `data.decodeU8()`.